### PR TITLE
serve ui with figwheel

### PR DESCRIPTION
**Pull request by zarsky-broad**
JIRA Issue: N/A

Corresponds to https://github.com/broadinstitute/firecloud-develop/pull/156.

@dmohs and I had a conversation, and it occurred to us that it was confusing and unnecessary to be serving the UI app with apache and hot reloading with figwheel when figwheel has a server built-in.
- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Make sure Swagger is updated if API changes
- [x] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [x] **Submitter**: JIRA ticket checks:
  - Acceptance criteria exists and is met
  - Note any changes to implementation from the description
  - Add notes on what you've tested
- [x] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [x] **Submitter**: Update FISMA documentation if changes to:
  - Authentication
  - Authorization
  - Encryption
  - Audit trails
- [x] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [x] Tell the tech lead (TL) that the PR exists if they wants to look at it
- [x] Anoint a lead reviewer (LR). **Assign PR to LR**
- Review cycle:
  - LR reviews
  - Rest of team may comment on PR at will
  - **LR assigns to submitter** for feedback fixes
  - Submitter rebases to develop again if necessary
  - Submitter makes further commits. DO NOT SQUASH
  - Submitter updates documentation as needed
  - Submitter **reassigns to LR** for further feedback
- [x] **TL** sign off
- [x] **LR** sign off
- [x] **Assign to submitter** to finalize
- [x] **Submitter**: Verify all tests go green, including CI tests
- [x] **Submitter**: Squash commits and merge to develop
- [x] **Submitter**: Delete branch after merge
- [x] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [x] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [x] **Submitter**: Inform other teams of any API changes via hipchat and/or email
- [x] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: project.clj [hunk 1]
```diff
@@ -12,12 +12,14 @@
...
-      {:websocket-url (str "ws://" host ":3449/figwheel-ws")})))
+      {:websocket-url (str "ws://" host ":80/figwheel-ws")})))
 
 
 (defn- get-figwheel-server-opts []
   (when (inside-container?)
-    {:hawk-options {:watcher :polling}}))
+    {:http-server-root ""
+    :server-port 80
+    :hawk-options {:watcher :polling}}))
```
Comment (MattPutnam) [inline project.clj]: nit: indentation

**Comment by MattPutnam** [comment]
👍 seems legit

File: project.clj (hunk 1 above)
Comment (zarsky-broad) [reply project.clj]: Fixed.

**Comment by davidangb** [comment]
I'm open to debate on this, but I disagree with the principle. Because we serve the UI via apache in prod (and dev, alpha, etc), I think we should serve the UI via apache in our dev envs. This ensures that during development, we see the exact same behavior that our end users will see, including various apache configs.

The counter-argument is that we want to make our dev envs as frictionless/lightweight as possible, so we accept the potential differences. There are other ways to run via apache in dev envs.

Does this (coupled with broadinstitute/firecloud-develop#156) change any behavior when running via the full-stack firecloud-docker-compose.yaml? I can't imagine it does ...

**Comment by dmohs** [comment]
@davidangb You raise a good point and it spurred a discussion. I'll summarize our thoughts as best I can.

There is value both in making development lightweight and in ensuring we are committing code that is as likely as possible to work in all of our deployment environments.

How about we say that developers should generally do a final round of testing by building the container before deploying so they can catch issues related to how the software will work in the deployed environment?

**Comment by davidangb** [comment]
sounds good to me. As long as the full-stack docker-compose still serves via container, I think we have a good set of options.

If it turns out that UI-served-by-figwheel and UI-served-by-apache have nontrivial differences, then we can always roll this back.

**Pull request merged**